extensions/channels.py

```diff
@@ -10,23 +10,16 @@ def __init__(self, bot):
 
     def channel_is_thread():
         async def predicate(ctx):
-            conn = ctx.bot.conn
-            c = conn.cursor()
-            c.execute("SELECT * FROM threads WHERE channel_id = ?", (ctx.channel.id,))
-            if c.fetchone() is None:
+            if ctx.bot.database.get_thread(ctx.channel.id) is None:
                 raise errors.ThreadOnly(f"This command may only be used in a channel that was created using the `{ctx.bot.command_prefix}thread` command.")
             return True
         return commands.check(predicate)
 
     def author_can_archive():
         # Only raises an error if the user isn't the thread author and doesn't have the Trustee role. If you want an error message for not being in a thread, also use channel_is_thread.
         async def predicate(ctx):
-            conn = ctx.bot.conn
-            c = conn.cursor()
-            c.execute("SELECT author_id FROM threads WHERE channel_id = ?", (ctx.channel.id,))
             role = discord.utils.find(lambda r: r.name==config.trustee_role, ctx.author.roles)
-            row = c.fetchone()
-            if row is not None and ctx.author.id!=row[0] and role is None:
+            if ctx.bot.database.get_author_of_thread(ctx.channel.id) is not None and role is None:

```

Comment:
The logic here is incorrect.
```suggestion
            thread_author_id = ctx.bot.database.get_author_of_thread(ctx.channel.id)
            if (thread_author_id is None or ctx.author.id!=thread_author_id) and role is None:
```

Reply:
Ah, I had some problems understanding it all because originally it was `if row is not None and ctx.author.id!=row[0] and role is None:` as opposed to `(thread_author_id is None or ctx.author.id!=thread_author_id) and role is None:`.